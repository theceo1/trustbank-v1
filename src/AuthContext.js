import React, { createContext, useContext, useState } from 'react';

const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  const login = async ({ email, password }) => {
    // Simulate an API call for login
    if (email === 'testing4@test.com' && password === 'password') {
      setUser({ email });
    } else {
      throw new Error('Invalid credentials');
    }
  };

  const logout = () => {
    setUser(null);
  };

  return (
    <AuthContext.Provider value={{ user, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => {
  return useContext(AuthContext);
};
