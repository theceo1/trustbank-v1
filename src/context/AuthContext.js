// src/context/AuthContext.js
import React, { createContext, useState, useContext, useEffect } from 'react';
import { useRouter } from 'next/router';

const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const router = useRouter();

  useEffect(() => {
    // Load user from local storage or API
    const loggedInUser = localStorage.getItem('user');
    if (loggedInUser) {
      setUser(JSON.parse(loggedInUser));
    }
  }, []);

  const login = (userData) => {
    setUser(userData);
    localStorage.setItem('user', JSON.stringify(userData));
    router.push('/dashboard');
  };

  const logout = () => {
    setUser(null);
    localStorage.removeItem('user');
    router.push('/login');
  };

  return (
    <AuthContext.Provider value={{ user, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => useContext(AuthContext);
