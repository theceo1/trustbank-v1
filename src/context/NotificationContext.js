import React, { createContext, useContext, useState, useCallback } from 'react';
import { v4 as uuidv4 } from 'uuid';

const NotificationContext = createContext();

export const useNotification = () => useContext(NotificationContext);

export const NotificationProvider = ({ children }) => {
  const [notifications, setNotifications] = useState([]);

  const addNotification = useCallback((type, message) => {
    const id = uuidv4();
    setNotifications((prev) => [...prev, { id, type, message }]);
    setTimeout(() => {
      setNotifications((prev) => prev.filter((notification) => notification.id !== id));
    }, 5000);
  }, []);

  return (
    <NotificationContext.Provider value={{ notifications, addNotification }}>
      {children}
    </NotificationContext.Provider>
  );
};
