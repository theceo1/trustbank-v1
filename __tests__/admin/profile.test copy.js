import React from 'react';
import { render } from '@testing-library/react';
import Profile from '@/pages/profile';
import { AuthProvider } from '@/context/AuthContext';

const renderWithAuthProvider = (ui, { providerProps, ...renderOptions }) => {
  return render(
    <AuthProvider {...providerProps}>{ui}</AuthProvider>,
    renderOptions
  );
};

test('renders Profile component', () => {
  const { asFragment } = renderWithAuthProvider(<Profile />, {
    providerProps: { value: { user: { name: 'Test User' } } },
  });
  expect(asFragment()).toMatchSnapshot();
});
