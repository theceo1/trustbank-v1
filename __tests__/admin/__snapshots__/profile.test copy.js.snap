// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile component 1`] = `
<DocumentFragment>
  <div
    class="container mx-auto py-8 px-4"
  >
    <h1
      class="text-3xl font-bold mb-4"
    >
      Profile
    </h1>
    <div
      class="text-gray-700 dark:text-gray-300 mb-8"
    >
      <img
        alt="profile"
        class="rounded-full"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fuser%2Fuser-06.png&w=640&q=75"
        srcset="/_next/image?url=%2Fimages%2Fuser%2Fuser-06.png&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fuser%2Fuser-06.png&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
      <h2
        class="text-2xl"
      >
        Danish Heilium
      </h2>
      <p>
        UI/UX Designer
      </p>
    </div>
    <a
      class="text-teal-500 hover:underline"
      href="/dashboard"
    >
      Go To Dashboard
    </a>
  </div>
</DocumentFragment>
`;
